framework/include/base/MooseFunctor.h

```diff
@@ -139,6 +153,23 @@ struct FaceArg
   {
     return {fi->neighborPtr(), fi, correct_skewness, apply_gradient_to_skewness, neighbor_sub_id};
   }
+
+  friend bool operator<(const FaceArg & l, const FaceArg & r)
+  {
+    return std::make_tuple(l.fi,
+                           l.limiter_type,
+                           l.elem_is_upwind,
+                           l.correct_skewness,
+                           l.apply_gradient_to_skewness,
+                           l.elem_sub_id,
+                           l.neighbor_sub_id) < std::make_tuple(r.fi,
+                                                                r.limiter_type,
+                                                                r.elem_is_upwind,
+                                                                r.correct_skewness,
+                                                                r.apply_gradient_to_skewness,
+                                                                r.elem_sub_id,
+                                                                l.neighbor_sub_id);
```

Comment:
just leaving a note that the ordering of the tuple might matter for how the map is sorted, which will matter for performance. 

Looking at this now is premature though